src/org/parosproxy/paros/network/SSLConnector.java

```diff
@@ -529,17 +534,29 @@ public Socket createSocket(Socket socket, String host, int port,
 	 * @throws IOException
 	 */
 	public Socket createTunnelServerSocket(String targethost, Socket socket) throws IOException {
+		InetAddress listeningAddress = socket.getLocalAddress();
 		// ZAP: added host name parameter
-		SSLSocket s = (SSLSocket) getTunnelSSLSocketFactory(targethost).createSocket(socket, socket
+		SSLSocket s = (SSLSocket) getTunnelSSLSocketFactory(targethost, listeningAddress).createSocket(socket, socket
 				.getInetAddress().getHostAddress(), socket.getPort(), true);
 		
 		s.setUseClientMode(false);
 		s.startHandshake();
 		return s;
 	}
 
-	// ZAP: added new ServerSocketFaktory with support of dynamic SSL certificates
+
+	/**
+	 * @deprecated (TODO add version) No longer used/needed.
+	 *
+	 * ZAP: added new ServerSocketFactory with support of dynamic SSL certificates

```

Comment:
Better just remove it (it should not be in the JavaDoc, also there's already related ZAP comments in the license header).

Reply:
Ok that's partially my fault, I think I suggested moving it here earlier. I didn't check the header comments ...